src/wallet-account.ts

```diff
@@ -106,6 +109,25 @@ export class WalletConnection {
         return !!this._authData.accountId;
     }
 
+    /**
+     * Returns promise of completing singing in
+     * @example
+     * ```js
+     * // on login callback page
+     * const wallet = new WalletConnection(near, 'my-app');
+     * wallet.isSignedIn(); // false
+     * await wallet.promiseSignIn();
+     * wallet.isSignedIn(); // true
+     * ```
+     */
+    async promiseSignIn() {
+        if (!this._completeSignInPromise) {
+            return {};
+        }
+
+        return this._completeSignInPromise;
+    }
```

Comment:
Do we need an additional method here? Could we just await the value we assigned the promise to in the constructor? e.g. from test:

```javascript
const newWalletConn = new nearApi.WalletConnection(nearFake, 'promise_on_complete_signin');

await newWalletConn._completeSignInPromise;
expect(await keyStore.getKey('networkId', 'near2.account')).toEqual(keyPair);
...
```

Reply:
I don't think you'd want to expose the internals that way. Like if you were to change the impl, you'd be locking people in to that.

Reply:
Good point. To address that we can change this to a getter, there's no need for the check and object return

Reply:
We want this to be a predicate that resolves to whether sign in was completed on this load of the instance. Therefore the return value should not be the same in both cases. We don't imply failure by returning false if we get the naming right. 

We can actually create a second method here called `isSignedInAsync` to complement the already existing `isSignedIn`:

```javascript
async isSignedInAsync() {
       if (!this._completeSignInPromise) {
           return this.isSignedIn();
       }

       await this._completeSignInPromise;
       return this.isSignedIn();
}
```